Combine Drone release steps

silverwind opened the pull request:
See https://github.com/go-gitea/gitea/issues/9324. I don't know if this will fix the issue but it seems worth a try to land this on master just to watch the release-master task run. If it does not work out, we can revert it.

lafriks commented:
I don't see how this does change things as checked out directory are shared between steps except if there are not in the same pipeline. Also imho js/css build step should be in different drone step

silverwind commented:
Yes, they should be shared but the only logical explanation for https://github.com/go-gitea/gitea/issues/9324 is that they aren't for some reason.

I do think it should be possible to produce a working release locally via `make release` and not only via CI, so I see the `release` task dependencies as must-haves.

techknowlogick commented:
So embedding works (we can see this with vendor'd css/js, as well as templates are embedded, this means go generate with bindata env is working.

I'm hesitant to merge this without testing, luckily drone allows us to run specific pipelines via CLI. I will test this change via CLI (which will allow me to also put in some debugging into drone that'll show more info (ex. `ls` of a specific dir before running `make release`)

zeripath commented:
This change means we lose testing on node10 doesn't it?

techknowlogick commented on .drone.yml:
```diff
@@ -389,16 +389,11 @@ steps:
       - git fetch --tags --force
 
   - name: build
```
```suggestion
   - name: static
```
There are steps below that depend on this being the step name

techknowlogick commented on .drone.yml:
```diff
@@ -497,16 +492,11 @@ steps:
       - git fetch --tags --force
 
   - name: build
```
```suggestion
   - name: static
```

techknowlogick commented:
@zeripath nope, this is just for the release steps, the node10 container is still in the testing pipeline.

axifive commented:
@techknowlogick Can you also debugging already existing step in master? To understand why it's not worked, maybe this can help avoid similar errors in the future.

techknowlogick commented:
Oh, I know what's going on! I can't believe I missed it. The build & static step are run at the same time, this means that even though the files exist where they should, the go generate command ends up being run before the JS/CSS files are finished being generated. the "depends_on" needs to be removed from each step in the release pipelines to force each step to be run in sequence.

silverwind commented:
@techknowlogick thanks for finding it. It's now clear to me that as soon as a `depends_on` is introduced, drone will parallelize (https://docker-runner.docs.drone.io/configuration/parallelism/). I went ahead and removed all `depends_on` in the release pipelines so they should now always serialize.

silverwind commented:
Squashed and added new commit message.

silverwind commented:
One more tweak pushed which make `js css` dependencies of `generate`. This fixes the case where`make generate build` (which some people have in their build scripts) would result in `generate` running only once and too early.

silverwind commented:
One more tweak: Removed useless `go-all` target. Now `go-check` only runs once instead of twice and before `generate` which also depends on `go` present.

The `go` target itself could potentially also removed but I kept it because it allows for quicker building without `generate`. Thought, the same could also be achieved using `make gitea`.

silverwind commented:
`go` target removed as well for simplicity's sake. One can still run `make gitea` (`make gitea.exe` on Windows) to achieve the same.

techknowlogick commented:
Thanks @silverwind for all of your work on this :)
